cartography/intel/okta/oktaintel.py

```diff
@@ -0,0 +1,1193 @@
+        user_props["okta_last_updated"] = okta_user.lastUpdated.strftime("%m/%d/%Y, %H:%M:%S")
+    else:
+        user_props["okta_last_updated"] = None
+
+    if okta_user.passwordChanged:
+        user_props["password_changed"] = okta_user.passwordChanged.strftime("%m/%d/%Y, %H:%M:%S")
+    else:
+        user_props["password_changed"] = None
+
+    if okta_user.transitioningToStatus:
+        user_props["transition_to_status"] = okta_user.transitioningToStatus
+    else:
+        user_props["transition_to_status"] = None
+
+    return user_props
+
+
+def _load_okta_users(neo4j_session, okta_org_id, user_list, okta_update_tag):
+    """
+    Load Okta user information into the graph
+    :param neo4j_session: session with neo4j server
+    :param okta_org_id: oktat organization id
+    :param user_list: list of users
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+
+    ingest_statement = """
+    MATCH (org:OktaOrganization{id: {ORG_ID}})
+    WITH org
+    UNWIND {USER_LIST} as user_data
+    MERGE (new_user:OktaUser{id: user_data.id})
+    ON CREATE SET new_user.firstseen = timestamp()
+    SET new_user.first_name = user_data.first_name,
+    new_user.last_name = user_data.last_name,
+    new_user.login = user_data.login,
+    new_user.email = user_data.email,
+    new_user.second_email = user_data.second_email,
+    new_user.mobile_phone = user_data.mobile_phone,
+    new_user.created = user_data.created,
+    new_user.activated = user_data.activated,
+    new_user.status_changed = user_data.status_changed,
+    new_user.last_login = user_data.last_login,
+    new_user.okta_last_updated = user_data.okta_last_updated,
+    new_user.password_changed = user_data.password_changed,
+    new_user.transition_to_status = user_data.transition_to_status,
+    new_user.lastupdated = {okta_update_tag}
+    WITH new_user, org
+    MERGE (org)-[org_r:RESOURCE]->(new_user)
+    ON CREATE SET org_r.firstseen = timestamp()
+    SET org_r.lastupdated = {okta_update_tag}
+    WITH new_user
+    MERGE (h:Human{email: new_user.email})
+    ON CREATE SET new_user.firstseen = timestamp()
+    SET h.lastupdated = {okta_update_tag}
+    MERGE (h)-[r:IDENTITY_OKTA]->(new_user)
+    ON CREATE SET new_user.firstseen = timestamp()
+    SET h.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest_statement,
+        ORG_ID=okta_org_id,
+        USER_LIST=user_list,
+        okta_update_tag=okta_update_tag,
+    )
+
+
+def _sync_okta_users(neo4j_session, okta_org_id, okta_update_tag):
+    """
+    Sync okta users
+    :param neo4j_session: Session with Neo4j server
+    :param okta_org_id: Okta organization id to sync
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+
+    logger.debug("Syncing Okta users")
+    user_client = _create_user_client(okta_org_id)
+    data = _get_okta_users(user_client)
```

Comment:
As mentioned above, please follow the pattern established in other intel modules where the raw data is retrieved from the API, then passed to transform by the sync function, then passed to the load function.

Reply:
This applies to all the _sync functions below as well.